make.md

```diff
@@ -0,0 +1,97 @@
+
+# Continuous Integration with make
+Continuous integration of the workflows repository automates the jobs to deploy the workflows in the target cluster:
+* Build the containerized image of the workflow and the related applications
+* Push the containerized image to an image registry
+* Generate the K8s manifests for the workflow
+* Customize the manifests according to the selected deployment option (either `kustomize` or `helm`)
+* Commit the manifests and the deployment changes to a configurable deployment repository
+
+Generating the PR for the pushed commit is out of the scope of the make procedure, as it depends
+on the specific git service provider (e.g. GitHib rather than Bitbucket).
+
+## Prerequisites
+* `make` version 3.8x
+* `docker` or `podman` containers
+* Linux OS
+* Container registry credentials to publish the generated image
+
+The actuall implementation is delegated to shell scripts localted under the [scripts](./scripts/) folder that run
+in a containerized image including all the required dependencies: there's no need to install any other 
+tool in the local environment. 
+
+## Configuration variables
+Variables can be used to configure the behavior of the [Makefile](./Makefile):
+
+| Variable | Description | Default |
+|----------|-------------|---------|
+| WORKFLOW_ID | The workflow ID | **Must be provided** |
+| CONTAINER_ENGINE | The container engine | `podman` if available, otherwise `docker` |
+| WORKDIR | The working directory where the repo files are copied | Execution of `mktemp -d` |
+| GIT_USER_NAME | Git username | Username from `git config` |
+| GIT_USER_EMAIL | Git user email | User email from `git config` |
+| PR_OR_COMMIT_URL | URL to use in commit message | `<remote URL>/commits/<commit ID>` from local repo |
+| GIT_TOKEN | Git credentials token to push the commit | Content of `.git_token` file |
```

Comment:
I like how my added comments that are being resolved only by reading the PR till its end :-)

Reply:
I always do the same mistake, start commenting before reaching the end 🤭